### ISSUE TITLE
Configurable distance metrics

### ISSUE THREAD

**Issue by dekuenstle** [dekuenstle/cblearn#34]
Where ever we use distance metrics (usually Euclidean), make this configurable.

Default euclidean. Allow string or function argument and additional parameters.+
Minimum: Minkowski distance, at best every metric of pdist (in sklearn or pytorch).

Evaluate, when this works well and when not.

**Comment by dekuenstle** [comment]
It might be worth to implement a basic routine for supported distance metrics, that is available in the whole library.
Support both evaluation orders, scipy and torch, and a basic set of metrics (e.g. Minkowski): 
* X -> Distance Matrix -> Triplet-Distances (high memory usage)
* X -> X-Triplets -> Triplet-Distances (slower)
